Content.Server/Tesla/EntitySystem/TeslaEnergyBallSystem.cs

```diff
@@ -0,0 +1,51 @@
+using Content.Server.Administration.Logs;
+using Content.Server.Singularity.Components;
+using Content.Server.Tesla.Components;
+using Content.Shared.Database;
+using Content.Shared.Singularity.Components;
+using Content.Shared.Mind.Components;
+using Content.Shared.Tag;
+using Robust.Shared.Physics.Events;
+using Content.Server.Lightning.Components;
+using Robust.Server.Audio;
+using Content.Server.Singularity.Events;
+
+namespace Content.Server.Tesla.EntitySystems;
+
+/// <summary>
+/// A component that tracks an entity's saturation level from absorbing other creatures by touch, and spawns new entities when the saturation limit is reached.
+/// </summary>
+public sealed class TeslaEnergyBallSystem : EntitySystem
+{
+    [Dependency] private readonly AudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TeslaEnergyBallComponent, EntityConsumedByEventHorizonEvent>(OnConsumed);
+    }
+
+    private void OnConsumed(Entity<TeslaEnergyBallComponent> tesla, ref EntityConsumedByEventHorizonEvent args)
+    {
+        Spawn(tesla.Comp.ConsumeEffectProto, Transform(args.Entity).Coordinates);
+        if (TryComp<SinguloFoodComponent>(args.Entity, out var singuloFood))
+        {
+            AdjustEnergy(tesla, tesla.Comp, singuloFood.Energy);
+        } else
```

Comment:
SinguloFoodComponent is already handled elsewhere.

Reply:
Here you have to get energy just from absorbing any object, not just singulofood.